lib/source/file.js

```diff
@@ -0,0 +1,167 @@
+/* global Log, Config */
+'use strict';
+
+const EventEmitter = require('events').EventEmitter;
+const fs = require('fs');
+
+class FileParser {
+  constructor() {
+    this.properties = Object.create(null);
+  }
+
+  /**
+   * Parse the file contents
+   * @param {Buffer} data
+   */
+  update(data) {
+    let properties = {};
+
+    try {
+      properties = JSON.parse(data.toString());
+    } catch (e) {
+      // We should have an error condition but tbd
+    }
+    this.properties = properties.properties;
+  }
+}
+
+class File extends EventEmitter {
+  constructor(options) {
+    super();
+    this.configure(options);
+
+    this.type = 'file';
+    this.name = this.path;
+    this.properties = Object.create(null);
+    this._okay = false;
+    this._updated = null;
+    this._parser = new FileParser();
+    this.service = null;
+  }
+
+  /**
+   * Configures the file plugin
+   *
+   * The following options are valid:
+   * - path (String) The path to the file
+   *
+   * @param {Object} options Options that can be set for the plugin
+   */
+  configure(options) {
+    const params = options || Object.create(null);
+
+    if (!params.path) {
+      throw new Error('No path supplied');
+    }
+
+    try {
+      fs.accessSync(params.path, fs.R_OK);
+      this.path = params.path;
+    } catch (err) {
+      this._error(err);
+      throw err;
+    }
+  }
+
+  /**
+   * Clear the underlying properties object
+   */
+  clear() {
+    this.properties = Object.create(null);
+  }
+
+  initialize() {
+    if (this.service) {
+      return;
+    }
+
+    this._update(this.path);
+
+    this.service = fs.watch(this.path, {persistent: true, recursive: false}, () => {
+      this._update();
+    });
+
+    this._okay = true;
+    this.emit('startup');
+  }
+
+  /**
+   * Stop watching the file for changes
+   *
+   * @returns {File}
+   */
+  shutdown() {
+    if (!this.service) {
+      return this;
+    }
+
+    // Release the FSWatcher handle
+    this.service.close();
+
+    delete this.service;
+
+    this._okay = false;
+    this.emit('shutdown');
```

Comment:
Looks like there's a missing `return this;` e.g. chaining should work if `initialize` has been called.